src/myrtlespeech/data/batch.py

```diff
@@ -94,6 +94,16 @@ def seq_to_seq_collate_fn(
         targets.append(target)
         target_seq_lens.append(target_seq_len)
 
+    # Sort the samples within the batch
+    samples = [
+        (input, in_seq_len, target, target_seq_len)
+        for input, in_seq_len, target, target_seq_len in zip(
+            inputs, in_seq_lens, targets, target_seq_lens
+        )
+    ]
+    sorted_samples = sorted(samples, key=lambda s: -s[0].size(-1))
```

Comment:
Use the `reverse` flag rather than negating the key value? 

```suggestion
    sorted_samples = sorted(samples, key=lambda s: s[0].size(-1), reverse=True)
```

Reply:
Yes good point. Done.